lib/languages/dlgthingtalk/entity_utils.js

```diff
@@ -0,0 +1,71 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of Genie
+//
+// Copyright 2016-2020 The Board of Trustees of the Leland Stanford Junior University
+//           2018 Google LLC
+//
+// Author: Giovanni Campagna <gcampagn@cs.stanford.edu>
+//
+// See COPYING for details
+"use strict";
+
+
+function tokensEquals(one, two) {
+    if (one === two)
+        return true;
+    // FIXME stemming
+    //if (Stemmer.stem(one).equals(Stemmer.stem(two)))
+    //    return true;
```

Comment:
This function comes from almond-tokenizer (as you might notice from the Java-ish .equals()) through almond-dialog-agent. The FIXME was added when I copied it to almond-dialog-agent, because we didn't have pluralize or stemmer. This whole code needs to change anyway.